hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/AbstractAsyncSocket.java

```diff
@@ -137,21 +137,17 @@ public final void close(String reason, Throwable cause) {
         this.closeReason = reason;
         this.closeCause = cause;
 
-        if (cause == null) {
```

Comment:
I find the new logic much harder to reason about than the old logic. The flow of the old logic is very easy to understand. So decreasing the readability of code for the sake of satisfying some arbitrary rules, I do not think is a step in the right direction.